include/OpenSpaceToolkit/Astrodynamics/Trajectory/TrajectorySegment.hpp

```diff
@@ -0,0 +1,161 @@
+
+    /// @brief                  Get name.
+    /// @return                 Name of the segment.
+
+    String getName() const;
+
+    /// @brief                  Get event condition.
+    /// @return                 Event condition.
+
+    Shared<EventCondition> getEventCondition() const;
+
+    /// @brief                  Get dynamics.
+    /// @return                 Dynamics.
+
+    Array<Shared<Dynamics>> getDynamics() const;
+
+    /// @brief                  Get numerical solver.
+    /// @return                 Numerical solver.
+
+    NumericalSolver getNumericalSolver() const;
+
+    /// @brief                  Access event condition.
+    /// @return                 Event condition.
+
+    const Shared<EventCondition>& accessEventCondition() const;
+
+    /// @brief                  Access dynamics.
+    /// @return                 Dynamics.
+
+    const Array<Shared<Dynamics>>& accessDynamics() const;
+
+    /// @brief                  Access numerical solver.
+    /// @return                 Numerical solver.
+
+    const NumericalSolver& accessNumericalSolver() const;
+
+    /// @brief                  Solve the segment
+    ///
+    /// @param                  [in] aState Initial state for the segment.
+    /// @param                  [in] aDynamicsArray Array of dynamics. Defaults to an empty array.
+    /// @param                  [in] aNumericalSolver Numerical solver to be used. Defaults to undefined.
+    /// @param                  [in] maximumPropagationDuration Maximum duration for propagation. Defaults to 30 days.
+    /// @return                 A Solution representing the result of the solve.
+
+    Solution solve(
+        const State& aState,
+        const Array<Shared<Dynamics>>& aDynamicsArray = Array<Shared<Dynamics>>::Empty(),
+        const NumericalSolver& aNumericalSolver = NumericalSolver::Undefined(),
+        const Duration& maximumPropagationDuration = Duration::Days(30.0)
+    ) const;
+
+    /// @brief                  Print the segment.
+    ///
+    /// @param                  [in] anOutputStream An output stream
+    /// @param                  [in] (optional) displayDecorators If true, display decorators
+
+    void print(std::ostream& anOutputStream, bool displayDecorator = true) const;
+
+    /// @brief                  Create a coasting segment.
+    ///
+    /// @param                  [in] anEventConditionSPtr An event condition.
+    /// @param                  [in] aDynamicsArray Array of dynamics. Defaults to an empty array.
+    /// @param                  [in] aNumericalSolver Numerical solver to be used. Defaults to undefined.
+    /// @return                 A TrajectorySegment for coasting.
+
+    static TrajectorySegment Coast(
+        const Shared<EventCondition>& anEventConditionSPtr,
+        const Array<Shared<Dynamics>>& aDynamicsArray = Array<Shared<Dynamics>>::Empty(),
+        const NumericalSolver& aNumericalSolver = NumericalSolver::Undefined()
+    );
```

Comment:
Does it ever make sense to create a `Coast` arc without dynamics or a solver? I'm not quite seeing why these are optional. What am I missing?